image-rs/src/unpack.rs

```diff
@@ -81,6 +93,63 @@ pub fn unpack<R: io::Read>(input: R, destination: &Path) -> Result<()> {
     Ok(())
 }
 
+fn convert_whiteout(path: &Path, header: &Header, destination: &Path) -> Result<bool> {
+    // Handle whiteout conversion
+    let name = path
+        .file_name()
+        .unwrap_or_default()
+        .to_str()
+        .ok_or(anyhow!("Invalid unicode in whiteout path: {:?}", path))?;
+
+    if name.starts_with(WHITEOUT_PREFIX) {
+        let parent = path
+            .parent()
+            .ok_or(anyhow!("Invalid whiteout parent fo path: {:?}", path))?;
+
+        if name == WHITEOUT_OPAQUE_DIR {
+            let destination_parent = destination.join(parent);
+            xattr::set(destination_parent, "trusted.overlay.opaque", b"y")?;
+            return Ok(false);
+        }
+
+        let original_name = name
+            .strip_prefix(WHITEOUT_PREFIX)
+            .ok_or(anyhow!("Failed to strip whiteout prefix for: {}", name))?;
+        let original_path = parent.join(original_name);
+        let path = CString::new(format!(
+            "{}/{}",
+            destination.display(),
+            original_path.display()
+        ))?;
+
+        let ret = unsafe { libc::mknod(path.as_ptr(), libc::S_IFCHR, 0) };
```

Comment:
can we use the nix crate for mknod?

Reply:
I think we can and happy to do it. There is existing use of `libc::lutimes` and `libc::utimes` in `unpack.rs` too. @mkulke Do you want me to do this work in this PR or should I open a separate issue/PR dealing with moving to nix?

Reply:
ah, wasn't aware, in this case then let's stick with the unsafe blocks here and open a dedicated issue. I think there's is good reason to use the nix crate's abstraction, but maybe there were considerations I'm not aware of